From 6c8fe07bd95145401ecf442272a1ef021f26c1b6 Mon Sep 17 00:00:00 2001
From: Lin Fuzhen <fuzhen.lin@freescale.com>
Date: Thu, 14 Mar 2013 15:20:37 +0800
Subject: [PATCH 05/34] ENGR00254281: add support to build differnt target
 images

Introduce a simple way to build ubifs for nand device base on Android
build system.
For those boards support booting from both NAND and eMMC/SD, this patch
provide a way to build out nand images with just add a build option to
enable build ubifs or ext4.
e.g make BUILD_TARGET_FS=ubifs will build image for NAND device
      make BUILD_TARGET_FS=ext4 will build image for eMMC/SD
	  if don't add BUILD_TARGET_FS when build, the build system will
	  build image according default config in BoardConfig.mk file.

Signed-off-by: Lin Fuzhen <fuzhen.lin@freescale.com>

Conflicts:
	core/Makefile
---
 core/Makefile | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/core/Makefile b/core/Makefile
index 004ac7e..e661603 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -823,13 +823,21 @@ endif
 recovery_resources_private := $(strip $(wildcard $(TARGET_DEVICE_DIR)/recovery/res))
 recovery_resource_deps := $(shell find $(recovery_resources_common) \
   $(recovery_resources_private) -type f)
+
+ifeq ($(TARGET_USERIMAGES_USE_UBIFS),true)
+recovery_fstab := $(strip $(wildcard $(TARGET_DEVICE_DIR)/recovery_nand.fstab))
+else
 ifdef TARGET_RECOVERY_FSTAB
 recovery_fstab := $(TARGET_RECOVERY_FSTAB)
 else
 recovery_fstab := $(strip $(wildcard $(TARGET_DEVICE_DIR)/recovery.fstab))
 endif
+endif
+
+ifneq ($(BOARD_USES_FULL_RECOVERY_IMAGE),true)
 # Named '.dat' so we don't attempt to use imgdiff for patching it.
 RECOVERY_RESOURCE_ZIP := $(TARGET_OUT)/etc/recovery-resource.dat
+endif
 
 ifeq ($(recovery_resources_private),)
   $(info No private recovery resources for TARGET_DEVICE $(TARGET_DEVICE))
-- 
1.8.0

